main.c

```diff
@@ -398,6 +399,16 @@ void NotifyOfLock(int x11_fd) {
 int main(int argc, char **argv) {
   setlocale(LC_CTYPE, "");
 
+  int fd = GetIntSetting("XSS_SLEEP_LOCK_FD", -1);
+  if (fd >= 0) {
+    int flags=  fcntl(fd, F_GETFD);
+    if (flags == 1)
+      /* Handle error */;
```

Comment:
I think printing a warning in this and the next case of fcntl failure should be good. It's not critical, so we don't have to abort.